src/dtls_connection_builder.rs

```diff
@@ -11,19 +12,34 @@ use crate::{Certificate, DtlsConnector, Identity, Protocol, Result, SrtpProfile}
 /// - Enabling Server Name Indication (SNI)
 pub struct DtlsConnectorBuilder {
     pub(crate) identity: Option<Identity>,
+    pub(crate) psk_identity: Option<(Bytes, Bytes)>,
     pub(crate) srtp_profiles: Vec<SrtpProfile>,
     pub(crate) min_protocol: Option<Protocol>,
     pub(crate) max_protocol: Option<Protocol>,
     pub(crate) root_certificates: Vec<Certificate>,
     pub(crate) accept_invalid_certs: bool,
     pub(crate) accept_invalid_hostnames: bool,
     pub(crate) use_sni: bool,
+    pub(crate) cipher_list: Vec<String>,
 }
 
 impl DtlsConnectorBuilder {
     /// Sets the identity to be used for client certificate authentication.
     pub fn identity(&mut self, identity: Identity) -> &mut DtlsConnectorBuilder {
         self.identity = Some(identity);
+        self.psk_identity = None;
+        self
+    }
+
+    /// Sets identity/key for client PSK authentication.
+    ///
+    /// Defaults to None
+    ///
+    /// # Hint
+    /// You should specify one of the PSK_* ciphers, i.e. PSK-AES128-CCM8
```

Comment:
Maybe this can be moved into an enum value with different ciphers in the future. https://www.openssl.org/docs/man1.1.1/man1/ciphers.html. 

Tough, I am fine to keep it like this right now.

Reply:
There is no such enum in OpenSSL itself, so I decided to leave it as is, not to get the API more constrained than the underlying library. 

Another question - how about turning Identity to enum of CertificateIdentity and PSKIdentity?

Reply:
Yea, because it can be constraining we are indeed better off to allow string input. On the other side, if it would be an enum or something like that, it would be less error-prone. 

If Identity could be either a Certificate or PSK Identity, the enum idea sounds like a good feature.

Reply:
I will do that and ammend my pull request.

Reply:
Changes are committed. There is still some asymmetry - connector supports both Cert/PSK, but acceptor only cert. Later I can think about how to support PSK for acceptor as well.